docs/observability/query-historical-inferences.mdx

```diff
@@ -0,0 +1,376 @@
+---
+title: How to query historical inferences
+sidebarTitle: Query historical inferences
+description: Learn how to retrieve and filter historical inferences from the TensorZero Gateway.
+---
+
+You can query historical inferences to analyze model behavior, debug issues, export data for fine-tuning, and more.
+The [TensorZero UI](/deployment/tensorzero-ui) provides an interface to browse and filter historical inferences.
+You can also query historical inferences programmatically using the TensorZero Gateway.
+
+<Tip>
+
+You can find a [complete runnable example](https://github.com/tensorzero/tensorzero/tree/main/examples/docs/guides/observability/query-historical-inferences) of this guide on GitHub.
+
+</Tip>
+
+## Query historical inferences by ID
+
+<span style={{ display: "block" }}>
+  <Badge color="blue">HTTP</Badge> `POST /v1/inferences/get_inferences`
+</span>
+<span style={{ display: "block" }}>
+  <Badge color="orange">TensorZero SDK</Badge> `client.get_inferences(...)`
+</span>
+
+Retrieve specific inferences when you know their IDs.
+
+### Request
+
+<ParamField body="ids" type="string[]" required>
+  List of inference IDs (UUIDs) to retrieve.
+</ParamField>
+
+<ParamField body="function_name" type="string">
+  Filter by function name. Including this improves query performance since
+  `function_name` is the first column in the ClickHouse primary key.
+</ParamField>
+
+<ParamField body="output_source" type="string" default="inference">
+
+Source of the output to return:
+
+- `"inference"`: Returns the original model output
+- `"demonstration"`: Returns human-curated feedback output if available
+- `"none"`: Returns the inference without output
+
+</ParamField>
+
+<Accordion title="Example">
+<Tabs>
+<Tab title="TensorZero Python SDK">
+
+You can retrieve inferences by ID using the TensorZero Python SDK.
+
+```python
+from tensorzero import TensorZeroGateway
+
+t0 = TensorZeroGateway.build_http(gateway_url="http://localhost:3000")
+
+t0.get_inferences(ids=["00000000-0000-0000-0000-000000000000"])
+```
+
+</Tab>
+<Tab title="HTTP">
+
+You can retrieve inferences by ID using the HTTP API.
+
+```bash
+curl -X POST http://localhost:3000/v1/inferences/get_inferences \
+  -H "Content-Type: application/json" \
+  -d '{"ids": ["00000000-0000-0000-0000-000000000000"]}'
+```
+
+</Tab>
+</Tabs>
+</Accordion>
+
+### Response
+
+<ResponseField name="inferences" type="StoredInference[]">
+  <Expandable title="StoredInference properties" defaultOpen="true">
+    <ResponseField name="dispreferred_outputs" type="array">
+      Outputs marked as dispreferred via feedback.
+    </ResponseField>
```

Comment:
this will only be populated if you set `demonstration` and is primarily used for preference-based optimization (DPO)